internal/pkg/diagnostics/diagnostics.go

```diff
@@ -593,19 +594,21 @@ func RedactSecretPaths(mapStr map[string]any, errOut io.Writer) map[string]any {
 		fmt.Fprintln(errOut, "No output redaction: secret_paths attribute is not a list.")
 		return mapStr
 	}
-	cfg := ucfg.MustNewFrom(mapStr)
+	cfg := ucfg.MustNewFrom(mapStr, ucfg.PathSep("."))
 	for _, v := range arr {
 		key, ok := v.(string)
 		if !ok {
 			fmt.Fprintf(errOut, "No output redaction for %q: expected type string, is type %T.\n", v, v)
 			continue
 		}
 
-		if ok, _ := cfg.Has(key, -1, ucfg.PathSep(".")); ok {
+		if ok, err := cfg.Has(key, -1, ucfg.PathSep(".")); ok {
 			err := cfg.SetString(key, -1, REDACTED, ucfg.PathSep("."))
 			if err != nil {
 				fmt.Fprintf(errOut, "No output redaction for %q: %v.\n", key, err)
 			}
+		} else {
+			fmt.Fprintf(errOut, "Unable to find secret path %q for redaction: %v.\n", key, err)
```

Comment:
can't err be nil here?

Reply:
yes, if the path just does not exist it would a log similar to`Unable to find secret path "non.existent.key" for redaction: <nil>.`

Reply:
not the biggest fan of printing nils. On the contrary if ok is true but err is not nil (buggy library) we missed completely logging it, right?

Reply:
yes. the current implementation for `Has` makes that impossible though

Reply:
then if you have strong feeling on handling an err correctly (imo) and, then please revert your change and make it explicitly ignored.

Reply:
I've seperated out the key not found, from the error output